modules/phase_field/test/tests/KKS_system/lagrange_multiplier.i

```diff
@@ -0,0 +1,342 @@
+    Fj_names = 'F1 F2'
+    hj_names = 'h1 h2'
+    cj_names = 'c1 c2'
+    eta_i = eta1
+    args = 'eta2'
+  []
+  [ACInterface1]
+    type = ACInterface
+    variable = eta1
+    kappa_name = kappa
+  []
+  [multipler1]
+    type = MatReaction
+    variable = eta1
+    v = lambda
+    mob_name = L
+  []
+
+  # Kernels for the Lagrange multiplier equation
+  [mult_lambda]
+    type = MatReaction
+    variable = lambda
+    mob_name = 2
+  []
+  [mult_ACBulkF_1]
+    type = KKSMultiACBulkF
+    variable = lambda
+    Fj_names = 'F1 F2 '
+    hj_names = 'h1 h2 '
+    gi_name = g1
+    eta_i = eta1
+    wi = 0.2
+    mob_name = 1
+    args = 'c1 c2 eta2 '
+  []
+  [mult_ACBulkC_1]
+    type = KKSMultiACBulkC
+    variable = lambda
+    Fj_names = 'F1 F2'
+    hj_names = 'h1 h2'
+    cj_names = 'c1 c2'
+    eta_i = eta1
+    args = 'eta2 '
+    mob_name = 1
+  []
+  [mult_CoupledACint_1]
+    type = SimpleCoupledACInterface
+    variable = lambda
+    v = eta1
+    kappa_name = kappa
+    mob_name = 1
+  []
+  [mult_ACBulkF_2]
+    type = KKSMultiACBulkF
+    variable = lambda
+    Fj_names = 'F1 F2 '
+    hj_names = 'h1 h2 '
+    gi_name = g2
+    eta_i = eta2
+    wi = 0.2
+    mob_name = 1
+    args = 'c1 c2 eta1 '
+  []
+  [mult_ACBulkC_2]
+    type = KKSMultiACBulkC
+    variable = lambda
+    Fj_names = 'F1 F2'
+    hj_names = 'h1 h2'
+    cj_names = 'c1 c2'
+    eta_i = eta2
+    args = 'eta1 '
+    mob_name = 1
+  []
+  [mult_CoupledACint_2]
+    type = SimpleCoupledACInterface
+    variable = lambda
+    v = eta2
+    kappa_name = kappa
+    mob_name = 1
+  []
+
+  # Kernels for constraint equation eta1 + eta2 = 1
+  # eta2 is the nonlinear variable for the constraint equation
+  [eta2reaction]
+    type = MatReaction
+    variable = eta2
+    mob_name = 1
+  []
+  [eta1reaction]
+    type = MatReaction
+    variable = eta2
+    v = eta1
+    mob_name = 1
+  []
+  [one]
+    type = BodyForce
+    variable = eta2
+    value = -1.0
+  []
+
+  # Phase concentration constraints
+  [chempot12]
+    type = KKSPhaseChemicalPotential
+    variable = c1
+    cb = c2
+    fa_name = F1
+    fb_name = F2
+  []
+  [phaseconcentration]
+    type = KKSMultiPhaseConcentration
+    variable = c2
+    cj = 'c1 c2'
+    hj_names = 'h1 h2'
+    etas = 'eta1 eta2'
+    c = c
+  []
+[]
+
+[AuxKernels]
+  [Fglobal_total]
+    type = KKSMultiFreeEnergy
+    Fj_names = 'F1 F2 '
+    hj_names = 'h1 h2 '
+    gj_names = 'g1 g2 '
+    variable = Fglobal
+    w = 0.2
+    interfacial_vars = 'eta1  eta2 '
+    kappa_names      = 'kappa kappa'
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = NEWTON
+  petsc_options_iname = '-pc_type -sub_pc_factor_shift_type'
+  petsc_options_value = 'lu       nonzero'
+  l_max_its = 30
+  nl_max_its = 10
+  l_tol = 1.0e-4
+  nl_rel_tol = 1.0e-10
+  nl_abs_tol = 1.0e-11
+
+  num_steps = 35
+  dt = 10
+[]
+
+[Preconditioning]
+  active = 'full'
+  [full]
+    type = SMP
+    full = true
+  []
+  [mydebug]
+    type = FDP
```

Comment:
Any reason to include this? Looks like `active = 'full'` is setting the previous block to be active, so can we remove the `mydebug` block?

Reply:
that entire `[Preconditioning]` block is unneeded. NEWTON creates a full matrix by default now.